src/frontend/src/components/SearchBar/SearchBar.jsx

```diff
@@ -90,17 +90,23 @@ function CustomizedInputBase(props) {
   const classes = useStyles();
   const { searchText, onChangeHandler, onFilterChangeHandler, filter, onFormSubmit } = props;
 
+  const prepareParamsAndNavigate = (text, filter) =>
+    navigate(`/search?text=${encodeURIComponent(text)}&filter=${encodeURIComponent(filter)}`);
```

Comment:
One other change to make here: let's pass `true` for `options.replace` to `navigate()` so that it doesn't add history state, but replaces what's there (i.e., then the back button will work like you expect), see: 

https://www.gatsbyjs.com/docs/gatsby-link/#replace-history-during-programmatic-navigation

Reply:
Let me do a research and try it again.